cnudie/retrieve.py

```diff
@@ -193,6 +195,69 @@ def lookup(
     return lookup
 
 
+def in_memory_cache_component_descriptor_lookup_async(
```

Comment:
will it make sense to create separate modules (cnudie/retrieve + cnudie/retrieve_async) to make this more of a drop-in replacement? we should deduplicate Dataclasses / Typedefs, though